### PR TITLE
Add inline documentation for Decodable

### PR DESCRIPTION
**Pull request by gfontenot**

### CODE REVIEW

File: Argo/Types/Decodable.swift [hunk 1]
```diff
@@ -1,4 +1,28 @@
 public protocol Decodable {
+  /**
+    The type of object that will be decoded.
+
+    In order to work with the rest of Argo, this needs to be the same as `Self`.
+
+    You should rarely need to set this value yourself. The only time you might
+    need to set it is if you allow `Self` to be subclassed and don't mark
+    `decode` as `final`. In that case, you will need to explicitly set
```
Comment (tonyd256) [inline Argo/Types/Decodable.swift]: just the function needs to be final or the whole class?

**Comment by tonyd256** [comment]
:+1:

File: Argo/Types/Decodable.swift [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+
+  /**
+    Decode an object from JSON.
+
+    This is the main entry point for Argo. This function declares how the
+    conforming type should be decoded from JSON. Since this is a failable
+    operation, we need to return a `Decoded` type from this function.
+
+    - parameter json: The `JSON` representation of this object
+    - returns: A decoded instance of the `DecodedType`.
```
Comment (tonyd256) [inline Argo/Types/Decodable.swift]: new line above

File: Argo/Types/Decodable.swift (hunk 1 above)
Comment (gfontenot) [inline Argo/Types/Decodable.swift]: I took another stab at this paragraph if you want to take another look?

File: Argo/Types/Decodable.swift [hunk 3]
```diff
@@ -1,4 +1,30 @@
 public protocol Decodable {
+  /**
+    The type of object that will be decoded.
+
+    In order to work with the rest of Argo, this needs to be the same as `Self`.
+
+    You should rarely need to set this value yourself. The only reason you'd
+    need to set it manually is if you allow your `decode` function to be
+    overridden by a subclass (so either your `class` or `func` are not marked
```
Comment (tonyd256) [inline Argo/Types/Decodable.swift]: what about removing the parens and changing "so" to "and"?

**Comment by tonyd256** [comment]
:+1: perfect

**Pull request merged**